lib/gtl_formatter.rb

```diff
@@ -31,7 +77,8 @@ def self.format_cols(view, row, controller)
         "prov_type" => :service_template_format,
       },
     }
-
+    cols.push(format_icon_column(view, row, options[:clickable])) if VIEW_WITH_CUSTOM_ICON.include?(view.db)
+    record = listicon_item(view, row['id'])
```

Comment:
This one is causing the `SELECT  "vms".* FROM "vms" WHERE "vms"."id" = $1` +1 query.

Reply:
funny that we still have a file called `gtl_formatter.rb` after we remove grid and tile in this PR, but that can be cleaned up later.  :-)

Reply:
Yeah, 650 more occurences of "gtl" or "GTL", but the react rewrite will catch quite a few of them too. (we still have 3 "cfme" too, it may take time)

Reply:
we do need to have `record` set to be able to format column data for list view, this code is not new, this has been moved from https://github.com/ManageIQ/manageiq-ui-classic/pull/7189/files/130d9655185c65240b3f422c592fef158e4b0ee5#diff-55c5b7aecfb519d0e4880eaf2788eb6eL882

Reply:
@himdel I was able to eliminate setting of `record` for most part, but still need it set for 2 columns, one is when calling decorators to determine the Provider type icon https://github.com/ManageIQ/manageiq-ui-classic/pull/7189/files/130d9655185c65240b3f422c592fef158e4b0ee5#diff-38fde2e3fdd09ed7a61246f63ac72d86R149
and second is when adding an order button in the Service Template list view https://github.com/ManageIQ/manageiq-ui-classic/pull/7189/files/130d9655185c65240b3f422c592fef158e4b0ee5#diff-38fde2e3fdd09ed7a61246f63ac72d86R119

Reply:
@himdel any suggestions on ^ ?

Reply:
No good suggestions I'm afraid :(

We *could* go through the decorator code, figure out which decorators use which fields, ensure the relevant report has all the necessary fields, and then run the decorator code on `row` instead of `record` by turning row into OpenStruct and including the decorator in the eigenclass.

But that feels rattle brittle, and you're right that the code was already there.